src/dbnode/encoding/istream.go

```diff
@@ -113,26 +120,8 @@ func readBitsInWord(w uint64, numBits uint8) uint64 {
 	return w >> (64 - numBits)
 }
 
-// consumeBuffer consumes numBits in is.current.
-func (is *IStream) consumeBuffer(numBits uint8) uint64 {
-	res := readBitsInWord(is.current, numBits)
-	is.current <<= numBits
-	is.remaining -= numBits
-	return res
-}
-
-func (is *IStream) readWordFromStream() error {
-	current, bytes, err := is.r.Read64()
-	is.current = current
-	is.remaining = 8 * bytes
-	is.err = err
-
-	return err
-}
-
 // Reset resets the IStream.
 func (is *IStream) Reset(reader xio.Reader64) {
-	is.err = nil
```

Comment:
big whoops that isn't caught by tests...

Reply:
okay my long-term memory is nonexistent: it was intentional. I should have removed `err` from struct itself, too, since there's no need to duplicate the error that would be returned by the underlying reader *anyway*.